### PR TITLE
feat: run pdb2pqr in python (no CLI)

### LINKED ISSUE #405: feat: use as an API
Previously, pdb2pqr was only available as a CLI. If it was used in some pipeline and it errors out, it was not easy to debug because it was only possible to infer the bug from the logs.

Making it available as an API function will enable python pipelines to smoothly integrate pdb2pqr and when it errors it would be easier to fix and report issues.

### PR DESCRIPTION
**Pull request by kiyoon**
Closes #405 

Usage:

```python
from pdb2pqr import run_pdb2pqr

run_pdb2pqr(["-h"])
```

Changes:

- There already is `run_pdb2pqr` in `tests/common.py` function for tests. To make the naming consistent and not confusing, I changed it to `run_pdb2pqr_for_tests` just like another function in the module called `run_propka_for_tests`.
- `main_driver()` doesn't include `io.setup_logger()` call anymore. It's moved to `main()` instead for maximum flexibility.

### CODE REVIEW

**Comment by kiyoon** [comment]
1. I don't know why some tests failed?
2. Once you approve this, I will apply ruff formatting following #409 , so don't merge just yet please.

**Comment by kiyoon** [comment]
In this case, `from __future__ import annotations` is used because of `def run_pdb2pqr(args: Sequence[str | PathLike]):`

**Review by sobolevnrm — APPROVED** [review]

**Comment by sobolevnrm** [comment]
Thanks for the PR!

**Pull request merged**
